### PR TITLE
Update CLI to use latest FireFly Core

### PR DESCRIPTION
**Pull request by nguyer**
This PR adds a lot of stuff, but primarily it gets it caught up with FireFly core so that things should actually work again. Other notable changes include:
- IPFS configuration fixes (they actually swarm!)
- Automatic loading of contracts on first run for a stack
- Error checking and propagation throughout
- Corrected help and usage text on all commands
- Tweaked status messages throughout

### CODE REVIEW

File: cmd/logs.go [hunk 1]
```diff
@@ -27,31 +27,43 @@ import (
 
 // logsCmd represents the logs command
 var logsCmd = &cobra.Command{
-	Use:   "logs",
+	Use:   "logs <stack_name>",
```
Comment (awrichar) [inline cmd/logs.go:30]: Should probably be consistent about `[positional_arg]` or `<positional_arg>` in these descriptions

Comment (awrichar) [reply cmd/logs.go:30]: Oh I guess you're trying to show optional vs. required args?

Comment (awrichar) [reply cmd/logs.go:30]: Maybe use Unix style `ARG` vs `[ARG]`...

Comment (nguyer) [reply cmd/logs.go:30]: I think the convention is normally `<required_arg>` or `[optional_arg]`

File: cmd/remove.go [hunk 2]
```diff
@@ -32,20 +31,22 @@ var force bool
...
 
-		if !stacks.CheckExists(stackName) {
-			log.Fatalf("Stack '%s' does not exist!", stackName)
+		if exists, err := stacks.CheckExists(stackName); err != nil {
+			return err
+		} else if !exists {
+			return fmt.Errorf("stack '%s' does not exist", stackName)
 		}
 
 		if !force {
```
Comment (awrichar) [inline cmd/remove.go:52]: Should probably move the `force` bool into this function like you do with the flags for the "logs" function. I didn't know how to do it.
(doesn't need to be done now, just noting it for one of us)

File: cmd/reset.go [hunk 3]
```diff
@@ -38,32 +37,34 @@ This is useful for testing when you want to start with a clean slate
...
 		fmt.Println("WARNING: This will completely remove your all transactions and data from your FireFly stack. Are you sure you want to do that?")
 		result, err := prompt.Run()
 
 		if err != nil || strings.ToLower(result) != "y" {
-			fmt.Printf("Canceled.")
-			return
+			fmt.Printf("canceled")
 		} else {
-			fmt.Printf("Reseting FireFly stack '%s'... ", stackName)
+			fmt.Printf("reseting FireFly stack '%s'... ", stackName)
```
Comment (awrichar) [inline cmd/reset.go:63]: resetting

File: ff/main.go [hunk 4]
```diff
@@ -21,4 +21,7 @@ import (
 
 func main() {
 	cmd.Execute()
+
+	// contract := contracts.ReadCompiledContract("/Users/nguyer/.firefly/stacks/dev/contracts/Payment.json")
+	// contracts.PublishABI("http://localhost:5555", contract)
```
Comment (awrichar) [inline ff/main.go:26]: Is this here for a reason?

File: cmd/reset.go [hunk 5]
```diff
@@ -38,32 +37,34 @@ This is useful for testing when you want to start with a clean slate
...
 			os.RemoveAll(path.Join(stacks.StacksDir, stackName, "data"))
-			fmt.Println("done!")
+			fmt.Println("done")
```
[the remark this answers is not in the document]
Comment (nguyer) [reply cmd/reset.go:65]: Go does not like punctuation at the end of error logs, so I was just trying to be consistent about every type of status that we print out.

**Comment by awrichar** [comment]
Seems good to me. I'm not terribly well-versed on the internals yet, but local testing seems to work well.

File: cmd/reset.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (awrichar) [reply cmd/reset.go:65]: Yea, I like the consistency.

**Review by awrichar — APPROVED** [review]

**Pull request merged**